### PR TITLE
Implement label checks using builtins

### PR DESCRIPTION
**Pull request by zregvart**
Now image manifest and image config is fetched using the `ec.oci.image_manifest` and `ec.oci.blob` builtin EC CLI functions. This allows the policy to handle failures to fetch either as policy violations.

Reference: EC-454

### CODE REVIEW

**Comment by zregvart** [comment]
There is a slight implementation difference of note here: `org.opencontainers.image.base.digest` is not optional, I thought it would be better to have a deterministic policy, i.e. look at the exact parent image rather than at some (perhaps newer) parent image.

File: policy/release/labels.rego [hunk 1]
```diff
@@ -44,6 +45,47 @@ warn contains result if {
 	)
 }
 
+# METADATA
+# title: Inaccessible image manifest
+# description: >-
+#   The image manfest is not accessible.
```
Comment (simonbaird) [inline policy/release/labels.rego:51]:
```suggestion
#   The image manifest is not accessible.
```

File: policy/release/labels.rego [hunk 2]
```diff
@@ -146,15 +190,90 @@ deny contains result if {
...
+#     Check the provided authentication configuration and the credentials
+#     within it.
+#   collections:
+#   - redhat
+#
+deny contains result if {
+	manifest := ec.oci.image_manifest(input.image.ref)
+
+	name := manifest.annotations["org.opencontainers.image.base.name"]
+	digest := manifest.annotations["org.opencontainers.image.base.digest"]
```
Comment (simonbaird) [inline policy/release/labels.rego:210]: If these annotations are not present, do we expect we already produced a "required_labels" violation? That would make sense to me, but could consider a comment to mention that.

File: policy/release/labels.rego [hunk 3]
```diff
@@ -146,15 +190,90 @@ deny contains result if {
...
+#     within it.
+#   collections:
+#   - redhat
+#
+deny contains result if {
+	manifest := ec.oci.image_manifest(input.image.ref)
+
+	parent_name := manifest.annotations["org.opencontainers.image.base.name"]
+	parent_digest := manifest.annotations["org.opencontainers.image.base.digest"]
+	parent_ref = sprintf("%s@%s", [parent_name, parent_digest])
```
Comment (simonbaird) [inline policy/release/labels.rego:236]: Could consider moving these into a parent_ref_from_manifest (or something) helper rule to reduce repetition.

**Review by simonbaird — APPROVED** [review]

File: policy/release/labels.rego (hunk 3 above)
Comment (zregvart) [reply policy/release/labels.rego]: :+1: found a nice refactor that can be done here

File: policy/release/labels.rego (hunk 2 above)
Comment (zregvart) [reply policy/release/labels.rego]: `required_labels` should be independent of the parent labels correlation checks, if these are not present that means the image has no (recorded) parent

**Pull request merged**